src/AWSS3.jl

```diff
@@ -1062,15 +1238,32 @@ end
 s3_sign_url(a...; b...) = s3_sign_url(global_aws_config(), a...; b...)
 
 """
-    s3_nuke_bucket(bucket_name)
+    s3_nuke_bucket([::AbstractAWSConfig], bucket_name)
 
-This function is NOT exported on purpose. AWS does not officially support this type of action,
-although it is a very nice utility one this is not exported just as a safe measure against
-accidentally blowing up your bucket.
+Deletes a bucket including all of the object versions in that bucket. Users should not call
+this function unless they are certain they want to permanently delete all of the data that
+resides within this bucket.
+
+The `s3_nuke_bucket` is purposefully *not* exported as a safe guard against accidental
+usage.
 
 !!! warning
 
-    It will delete all versions of objects in the given bucket and then the bucket itself.
+    Permanent data loss will occur when using this function. Do not use this function unless
+    you understand the risks. By using this function you accept all responsibility around
+    any repercussions with the loss of this data.
+
+# API Calls
+
+- [`ListObjectVersions`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectVersions.html)
+- [`DeleteObject`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html)
+- [`DeleteBucket`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteBucket.html)
+
+# Permissions
+
+- [`s3:ListBucketVersions`](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html#amazons3-ListBucketVersions)
+- [`s3:DeleteObjectVersion`](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html#amazons3-DeleteObjectVersion)
```

Comment:
This also needs `s3:DeleteObject`?

Reply:
I'll check this

Reply:
So it appears that the answer is no. Even on buckets without versioning enabled we require `s3:DeleteObjectVersion` as that is the permission required to delete an object when passing in the version parameter.

Reply:
I'll add a little blurb about this being required even for non-versioned buckets.